src/core/server/models/story/story.ts

```diff
@@ -727,3 +742,306 @@ export async function retrieveStorySections(
   // the null entries, the resulting array can not contain null.
   return results.filter((section) => section !== null).sort() as string[];
 }
+
+export async function markStoryForArchiving(
+  mongo: Db,
+  tenantID: string,
+  id: string,
+  now: Date
+) {
+  const result = await collection(mongo).findOneAndUpdate(
+    {
+      id,
+      tenantID,
+      $and: [
+        { $or: [{ isArchiving: { $exists: false } }, { isArchiving: false }] },
+        { $or: [{ isArchived: { $exists: false } }, { isArchived: false }] },
```

Comment:
This can be more compactly written:

```suggestion
        { isArchiving: { $in: [null, false] } },
        { isArchived: { $in: [null, false] } },
```

Reply:
Thanks, will update!